### PR TITLE
chore: allow passing custom tmpdir

### PR DESCRIPTION
**Pull request by npty**
# Description

Fix `invalid cross-device link (os error 18)` when I'm trying to deploy sui contract on a docker container ~~by allow passing `SUI_TMPDIR` to override the value from `node-tmp` lib~~ by remove the tmp dependency and create a unique temporarily folder in the same level as the script.

Here's full error:
```bash
➜ node sui/deploy-contract.js deploy AxelarGateway --signers wallet

Wallet address: 0xe529f4c63e18f8a3e8d4c2f5814d5d8d03b4d459e98b8ac818bc36426e6f8fec

Wallet balance: 1000 SUI

node:internal/errors:983
  const err = new Error(message);
              ^

Error: Command failed: sui move build --dump-bytecode-as-base64 --path /config/workspace/axelar-contract-deployments/sui/move/axelar_gateway --install-dir /tmp/tmp-36320-CXfjkmzk6vIj
    at genericNodeError (node:internal/errors:983:15)
    at wrappedFn (node:internal/errors:537:14)
    at checkExecSyncError (node:child_process:890:11)
    at execSync (node:child_process:962:15)
    at TxBuilder.<anonymous> (/config/workspace/axelar-contract-deployments/node_modules/@axelar-network/axelar-cgp-sui/dist/tx-builder.js:213:95)
    at Generator.next (<anonymous>)
    at /config/workspace/axelar-contract-deployments/node_modules/@axelar-network/axelar-cgp-sui/dist/tx-builder.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/config/workspace/axelar-contract-deployments/node_modules/@axelar-network/axelar-cgp-sui/dist/tx-builder.js:4:12)
    at TxBuilder.getContractBuild (/config/workspace/axelar-contract-deployments/node_modules/@axelar-network/axelar-cgp-sui/dist/tx-builder.js:208:16) {
  status: 1,
  signal: null,
  output: [
    null,
    'Failed to build Move modules: Committing lock file\n' +
      '\n' +
      'Caused by:\n' +
      '    0: failed to persist temporary file: Invalid cross-device link (os error 18)\n' +
      '    1: Invalid cross-device link (os error 18).\n',
    ''
  ],
  pid: 36363,
  stdout: 'Failed to build Move modules: Committing lock file\n' +
    '\n' +
    'Caused by:\n' +
    '    0: failed to persist temporary file: Invalid cross-device link (os error 18)\n' +
    '    1: Invalid cross-device link (os error 18).\n',
  stderr: ''
}
```

### CODE REVIEW

File: src/tx-builder.ts [hunk 1]
```diff
@@ -258,10 +258,15 @@ export class TxBuilder {
         const tmpobj = tmp.dirSync({ unsafeCleanup: true });
```
Comment (milapsheth) [inline src/tx-builder.ts:258]: instead of using `tmp`, maybe just hardcode a tmp folder, and cleanup before reusing?

**Review by milapsheth — APPROVED** [review]

File: src/tx-builder.ts (hunk 1 above)
Comment (npty) [reply src/tx-builder.ts:258]: yeah, should have no problem with it

File: src/tx-builder.ts [hunk 2]
```diff
@@ -246,24 +246,40 @@ export class TxBuilder {
...
         packageName: string,
         moveDir: string = `${__dirname}/../move`,
     ): Promise<{ modules: string[]; dependencies: string[]; digest: Bytes }> {
         const emptyPackageId = '0x0';
         updateMoveToml(packageName, emptyPackageId, moveDir);
 
-        tmp.setGracefulCleanup();
-
-        const tmpobj = tmp.dirSync({ unsafeCleanup: true });
+        const { tmpdir, rm } = this.prepareMoveBuild();
```
Comment (milapsheth) [inline src/tx-builder.ts:272]:
```suggestion
        const { tmpdir, rmTmpDir } = this.prepareMoveBuild();
```

File: src/tx-builder.ts [hunk 3]
```diff
@@ -246,24 +246,40 @@ export class TxBuilder {
...
+     * - dir is the path to the temporary directory
+     * - rm is a function to remove the temporary directory
+     */
+    private prepareMoveBuild() {
+        const tmpdir = fs.mkdtempSync(`${__dirname}/.move-build-`);
+        const rm = () => fs.rmSync(tmpdir, { recursive: true });
+
+        return {
+            tmpdir,
+            rm,
```
Comment (milapsheth) [inline src/tx-builder.ts:261]:
```suggestion
            rmTmpDir,
```

File: src/tx-builder.ts [hunk 4]
```diff
@@ -246,24 +246,40 @@ export class TxBuilder {
...
 
         const { modules, dependencies, digest } = JSON.parse(
-            execSync(`sui move build --dump-bytecode-as-base64 --path ${path.join(moveDir, packageName)} --install-dir ${tmpobj.name}`, {
+            execSync(`sui move build --dump-bytecode-as-base64 --path ${path.join(moveDir, packageName)} --install-dir ${tmpdir}`, {
                 encoding: 'utf-8',
                 stdio: 'pipe', // silent the output
             }),
         );
 
+        rm();
```
Comment (milapsheth) [inline src/tx-builder.ts:281]:
```suggestion
        rmTmpDir();
```

**Review by milapsheth — APPROVED** [review]

**Pull request merged**